mteb/encoder_interface.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+from typing import Any, Protocol, runtime_checkable
+
+import numpy as np
+import torch
+
+
+@runtime_checkable
+class Encoder(Protocol):
```

Comment:
An example of a protocol. Only specified the interface we want to use from an encoder model.